cli/cmd/metrics_diagnostics_util.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"sync/atomic"
+	"time"
+
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// shared between metrics and diagnostics command
+type metricsResult struct {
+	pod       string
+	container string
+	metrics   []byte
+	err       error
+}
+type byResult []metricsResult
+
+func (s byResult) Len() int {
+	return len(s)
+}
+func (s byResult) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s byResult) Less(i, j int) bool {
+	return s[i].pod < s[j].pod || ((s[i].pod == s[j].pod) && s[i].container < s[j].container)
+}
+
+// getResponse makes a http Get request to the passed url and returns the response/error
+func getResponse(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes, nil
+}
+
+// getContainerMetrics returns the metrics exposed by a container on the passed in portName
+func getContainerMetrics(
+	k8sAPI *k8s.KubernetesAPI,
+	pod corev1.Pod,
+	container corev1.Container,
+	emitLogs bool,
+	portName string,
+) ([]byte, error) {
+	portForward, err := k8s.NewContainerMetricsForward(k8sAPI, pod, container, emitLogs, portName)
+	if err != nil {
+		return nil, err
+	}
+
+	defer portForward.Stop()
+	if err = portForward.Init(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error running port-forward: %s", err)
```

Comment:
if `err != nil`, don't we want to `return`?

Reply:
Yeah. Missed that. Thanks